Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -137,7 +165,15 @@ def Test_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
     recvdContentChecksum = md5.md5(Contents).hexdigest().upper()
     if recvdContentChecksum != ContentChecksum:
         # data is corrupt do not proceed further
-        LOG_ACTION.log(LogType.Error, 'Content received did not match checksum, exiting Test')
+        LOG_ACTION.log(LogType.Error, 'Content received did not match checksum, exiting Test for ' + ConfigID)
+        return [0]
+    if not os.path.exists(os.path.join(PREFIX_ETC, ConfigID)):
```

Comment:
If ConfigID is None here then this will raise an AttributeError. Better to verify this condition in regular TEST call below, after the init_vars kicks in.